[botan] add version 2.0.1

Mixaill opened the pull request:
Resolves #355

Mixaill committed:
[botan] add version 2.0.1. Resolves #355

msftclas commented:
@Mixaill,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by Microsoft. We will now review your pull request.
_Thanks,
Microsoft Pull Request Bot_

ras0219-msft committed:
[botan] Use out of source builds

ras0219-msft committed:
[botan] Cleanup.

Avoid installing a debug cli tool.
Rename the headers to just include/botan instead of include/botan-2/botan.
Install the tool's required DLLs, if any.
Add vcpkg distribution information to the build process.

alexkaratarakis commented:
I am unable to build this; jom.exe (1.1.2) is crashing and failing the build for me.
@ras0219-msft can you try building this?

ras0219-msft commented:
Thanks for the PR!

I made a few changes while merging:
- We use out-of-source-builds, so that we can get a from-scratch build every time (more robust)
- I removed the debug copy of `botan-cli.exe`
- The headers were installed to `include\botan-2\botan`. All of the Botan examples online use `#include <botan/XYZ.h>`, so I reorganized them to work out of the box.
- When building Botan DLLs, the tool requires access to the built DLL (applocal).
- While doing the above, I saw an option to provide distribution information, so I filled in "vcpkg ${TARGET_TRIPLET}"
